test/com/amazon/ion/impl/bin/IonEncoder_1_1Test.java

```diff
@@ -0,0 +1,235 @@
+package com.amazon.ion.impl.bin;
+
+import com.amazon.ion.IonType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+
+public class IonEncoder_1_1Test {
+
+    private static BlockAllocator ALLOCATOR = BlockAllocatorProviders.basicProvider().vendAllocator(11);
+    private WriteBuffer buf;
+    private ByteArrayOutputStream out;
+
+    @BeforeEach
+    public void setup() {
+        buf = new WriteBuffer(ALLOCATOR);
+        out = new ByteArrayOutputStream();
+    }
+
+    @AfterEach
+    public void teardown() {
+        buf = null;
+        out.reset();
+    }
+
+    private byte[] bytes() {
+        final ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try {
+            buf.writeTo(out);
+        } catch (final IOException e) {
+            throw new IllegalStateException(e);
+        }
+        return out.toByteArray();
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "     NULL, EA",
+            "     BOOL, EB 00",
+            "      INT, EB 01",
+            "    FLOAT, EB 02",
+            "  DECIMAL, EB 03",
+            "TIMESTAMP, EB 04",
+            "   STRING, EB 05",
+            "   SYMBOL, EB 06",
+            "     BLOB, EB 07",
+            "     CLOB, EB 08",
+            "     LIST, EB 09",
+            "     SEXP, EB 0A",
+            "   STRUCT, EB 0B",
+    })
+    public void testWriteNullValue(IonType value, String expectedBytes) {
+        int numBytes = IonEncoder_1_1.writeNullValue(buf, value);
+        String actualBytes = byteArrayToHex(bytes());
+        Assertions.assertEquals(expectedBytes, actualBytes);
+        Assertions.assertEquals((expectedBytes.length() + 1)/3, numBytes);
```

Comment:
I can see why it works, but this is a confusing line. I think it warrants a method like `byteLengthFromHex` or something with a comment that describes the requirements of the string format that make it work.